packages/react/src/components/FileUploader/FileUploaderDropContainer.js

```diff
@@ -0,0 +1,152 @@
+import React, { useCallback, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { settings } from 'carbon-components';
+import { keys, matches } from '../../internal/keyboard';
+import uniqueId from '../../tools/uniqueId';
+
+const { prefix } = settings;
+
+export default function FileUploaderDropContainer(props) {
+  const inputRef = useRef();
+  const {
+    accept,
+    className,
+    id,
+    disabled,
+    labelText,
+    multiple,
+    name,
+    onAddFiles,
+    role,
+    tabIndex,
+    ...other
+  } = props;
+  const uid = useRef(uniqueId()).current;
+  const [isActive, setActive] = useState(false);
+  const classes = classNames(`${prefix}--file__drop-container`, {
+    [`${prefix}--file__drop-container--drag-over`]: isActive,
+    [className]: className,
+  });
+  const handleChange = useCallback(
```

Comment:
Any reason for `useCallback` here?

Reply:
to avoid recreating the function on render and instead creating it when the watched props change

Reply:
Is this a hot codepath or something? Just curious what drove the design decision to optimize

Reply:
yeah I think this is a standard use case for React's built in performance optimization

Reply:
We shouldn’t just be using callbacks everywhere unless there’s a measured need for it. The concerns around defining a function in render are often overblown.

Reply:
the usage of callbacks here wasn't preemptive IIRC, but without being able to recall the results of our testing back then I'll just remove it at this point